src/nplinker/pairedomics/podp_antismash_downloader.py

```diff
@@ -25,81 +31,228 @@
+    if Path.exists(genome_status_file):
+        with open(genome_status_file) as f:
+            for line in csv.reader(f):
+                asobj = GenomeStatus(*line)
+                genome_status[asobj.original_id] = asobj
+
+    return genome_status
+
+
+def _get_best_available_genome_id(
+        genome_id_data: dict[str, str]) -> str | None:
+    """Get the best available ID from genome_id_data dict.
+
+    Args:
+        genome_id_data(dict): dictionary containing information
+        for each genome record present.
+
+    Returns:
+        str | None: ID for the genome, if present, otherwise None. 
+    """
 
-def _get_best_available_genome_id(genome_id_data):
     if 'RefSeq_accession' in genome_id_data:
         return genome_id_data['RefSeq_accession']
-    elif 'GenBank_accession' in genome_id_data:
+    if 'GenBank_accession' in genome_id_data:
         return genome_id_data['GenBank_accession']
-    elif 'JGI_Genome_ID' in genome_id_data:
+    if 'JGI_Genome_ID' in genome_id_data:
         return genome_id_data['JGI_Genome_ID']
 
     logger.warning(
-        'No known genome ID field in genome data: {}'.format(genome_id_data))
+        f'No known genome ID field in genome data: {genome_id_data}')
     return None
 
 
-def _ncbi_genbank_search(genbank_id, retry_time=5.0):
-    url = NCBI_LOOKUP_URL_NEW.format(genbank_id)
-    logger.debug('Looking up GenBank data for {} at {}'.format(
-        genbank_id, url))
-    resp = httpx.get(url, follow_redirects=True)
-
-    if resp.status_code == httpx.codes.OK:
-        # the page should contain a <dl> element with class "assembly_summary_new". retrieving
-        # the page seems to fail occasionally in the middle of lengthy sequences of genome
-        # lookups, so there might be some throttling going on. this will automatically retry
-        # the lookup if the expected content isn't found the first time
-        soup = BeautifulSoup(resp.content, 'html.parser')
-        # find the <dl> element with class "assembly_summary_new"
-        dl_element = soup.find('dl', {'class': 'assembly_summary_new'})
-        if dl_element is not None:
-            return dl_element
+def _ncbi_genbank_search(genbank_id: str,
+                         retry_times: int = 3) -> Tag | NavigableString | None:
 
-    logger.debug(
-        'NCBI lookup failed, status code {}. Trying again in {} seconds'.
-        format(resp.status_code, retry_time))
-    time.sleep(retry_time)
-    logger.debug('Looking up GenBank data for {} at {}'.format(
-        genbank_id, url))
-    resp = httpx.get(url, follow_redirects=True)
-    if resp.status_code == httpx.codes.OK:
-        soup = BeautifulSoup(resp.content, 'html.parser')
-        # find the <dl> element with class "assembly_summary_new"
-        dl_element = soup.find('dl', {'class': 'assembly_summary_new'})
-        if dl_element is not None:
-            return dl_element
+    url = NCBI_LOOKUP_URL_NEW.format(genbank_id)
```

Comment:
```suggestion
    url = NCBI_LOOKUP_URL.format(genbank_id)
```
I did not see an `OLD` one, so let's rename it. Please rename the other places too.